pylint/extensions/confusing_elif.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2021 Andreas Finkler <andi.finkler@gmail.com>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+import astroid
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class ConfusingConsecutiveElifChecker(BaseChecker):
+    """Checks if "elif" is used right after an indented block that finishes with "if" or "elif" itself."""
+
+    __implements__ = IAstroidChecker
+
+    name = "confusing-elif-checker"
+    priority = -1
+    msgs = {
+        "R5601": (
+            "Consecutive elif with differing indentation level, consider creating a function to separate the inner elif",
+            "confusing-consecutive-elif",
+            "Used when an elif statement follows right after an indented block which itself ends with if or elif. "
+            "It may not be ovious if the elif statement was willingly or mistakenly unindented. "
+            "Extracting the indented if statement into a separate function might avoid confusion and prevent errors.",
+        )
+    }
+
+    @check_messages("confusing-consecutive-elif")
+    def visit_if(self, node: astroid.If):
+        body_ends_with_if = isinstance(
+            node.body[-1], astroid.If
+        ) and self._has_no_else_clause(node.body[-1])
+        if node.has_elif_block() and body_ends_with_if:
```

Comment:
Why not combine all of them into the if statement?

Reply:
Hm. It would be a bit shorter, but in my opinion also a bit harder to understand. 
By extracting this part into a variable what this boolean represents and naming it appropriately, the if clause is more readable. 
If it's all compressed into the if statement I would probably have to do a double take at the code to understand what it does if I hadn't written it myself (or come back to it months later...).

Reply:
Might be just my opinion, but I think it would be just as easy to understand since you don't have to remember what `body_ends_with_if` was
```py
        if (
            node.has_elif_block()
            and isinstance(node.body[-1], astroid.If)
            and self._has_no_else_clause(node.body[-1])
        ):
            ...
```

Reply:
I like the named variable, it prevent a comment and explicit what the boolean is. I understand that better than ``isinstance(node.body[-1], astroid.If)``